### PR TITLE
Adding inputmode attribute

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

File: gaia-text-input.js [hunk 1]
```diff
@@ -148,6 +149,15 @@ module.exports = component.register('gaia-text-input', {
       }
     },
 
+    inputmode: {
+      get: function() { return this.els.input.getAttribute('x-inputmode'); },
+      set: function(value) {
+        if (!value) { return; }
+        this.els.inner.setAttribute('x-inputmode', value);
```
Comment (wilsonpage) [inline gaia-text-input.js]: Don't think this line is required.

File: gaia-text-input.js [hunk 2]
```diff
@@ -148,6 +149,15 @@ module.exports = component.register('gaia-text-input', {
       }
     },
 
+    inputmode: {
+      get: function() { return this.els.input.getAttribute('x-inputmode'); },
+      set: function(value) {
+        if (!value) { return; }
```
Comment (wilsonpage) [inline gaia-text-input.js]: If value is falsey, or not a recognised type we should `removeAttribute()`

Comment (albertopq) [inline gaia-text-input.js]: What do you mean by recognised? I don't think we should check a list here. The input controller will show the keyboard according to whatever the user sets there, in the same way they can do with the attribute via HTML. What do you think?

Totally agree on removing the attribute when the value is falsey.

File: gaia-text-input.js (hunk 1 above)
Comment (albertopq) [inline gaia-text-input.js]: Sorry, just copied/pasted the 'type' attribute. Let me check! Thanks!

File: gaia-text-input.js (hunk 2 above)
Comment (wilsonpage) [inline gaia-text-input.js]: OK happy not to perform check then, but let's make sure to remove it when falsy :)

**Pull request merged**